packages/manager/apps/web/client/app/private-database/whitelist/private-database-whitelist.controller.js

```diff
@@ -192,5 +207,132 @@ angular.module('App').controller(
         );
       }
     }
+
+    getWebhostingNetworkStatus() {
+      return this.whitelistService
+        .getWebhostingNetworkStatus(this.serviceName)
+        .then((res) => {
+          this.webhostingNetworkStatus = res;
+
+          this.isWebhostingNetworkDisplayOn = [
+            WEBHOSTING_NETWORK_STATUS.enabling,
+            WEBHOSTING_NETWORK_STATUS.enabled,
+          ].includes(this.webhostingNetworkStatus);
+
+          this.isWebhostingNewtworkChangeDisabled = [
+            WEBHOSTING_NETWORK_STATUS.enabling,
+            WEBHOSTING_NETWORK_STATUS.disabling,
+          ].includes(this.webhostingNetworkStatus);
+        })
+        .catch((err) => this.alerter.error(err));
+    }
+
+    onWebhostingNetworkStatusChange() {
+      switch (this.webhostingNetworkStatus) {
+        case WEBHOSTING_NETWORK_STATUS.disabled:
+          this.enableWebhostingNetwork(this.serviceName);
+          break;
+        case WEBHOSTING_NETWORK_STATUS.enabled:
+          this.disableWebhostingNetwork(this.serviceName);
+          break;
+        default:
+          return;
+      }
+
+      this.isWebhostingNewtworkChangeDisabled = true;
+      this.alerter.success(
+        this.$translate.instant(
+          'private_database_tabs_whitelist_webhosting_network_access_success',
+        ),
+        this.$scope.alerts.main,
+      );
+    }
+
+    handleWebhostingNetworkUpdate(state, message) {
+      this.getWebhostingNetworkStatus();
+      this.isWebhostingNewtworkChangeDisabled = false;
+      if (state === 'error') {
+        this.alerter.error(message, this.$scope.alerts.main);
+      }
+    }
+
+    /**
+     * Enable webhosting access to database
+     * @param  {string} serviceName
+     * @return {Promise}
+     */
+    enableWebhostingNetwork(serviceName) {
+      return this.$http
+        .post(
+          `${this.whitelistService.rootPath}/${this.whitelistService.swsProxypassPath}/${serviceName}/webhostingNetwork`,
+        )
+        .then((res) => {
+          const taskId = res.data.id;
+          this.pollWebhostingNetwork(serviceName, taskId, 'enable');
+        })
+        .catch((err) => {
+          this.$scope.$broadcast(
+            'privateDatabase.webhostingNetwork.enable.error',
+            err.data.message,
+          );
+        });
+    }
+
+    /**
+     * Disable webhosting access to database
+     * @param  {string} serviceName
+     * @return {Promise}
+     */
+    disableWebhostingNetwork(serviceName) {
+      return this.$http
+        .delete(
+          `${this.whitelistService.rootPath}/${this.whitelistService.swsProxypassPath}/${serviceName}/webhostingNetwork`,
+        )
```

Comment:
Same as here https://github.com/ovh/manager/pull/6099/files#r776751892

Reply:
As seen with @marie-j , we were asked to move this code from the service to the controller to avoid using the rootScope.